### ISSUE TITLE
Remove automatic registration of `Node`s

### ISSUE THREAD

**Issue by gnn** [oemof/oemof#341]
Right now, instantiated `Node`s are automatically added to the most recently instantiated `EnergySystem`. This should be changed. In v0.2.0, new `Node`s should by default not implicitly be added to any `EnergySystem`, but it should be somehow made explicit to which `EnergySystem` a new `Node` should be added, either by:

  * explicitly modifying `Node.registry`,
  * having a parameter on `Node` or
  * manually adding `Node`s to the `EnergySystem`.

**Comment by ckaldemeyer** [comment]
> Right now, instantiated Nodes are automatically added to the most recently instantiated EnergySystem. This should be changed. In v0.2.0, new Nodes should by default not implicitly be added to any EnergySystem, but it should be somehow made explicit to which EnergySystem a new Node should be added, either by:
> 
> explicitly modifying Node.registry,
> having a parameter on Node or
> manually adding Nodes to the EnergySystem.

:+1:  I absolutely support this as it removes more hidden magic ;-)

**Comment by uvchik** [comment]
> 👍 I absolutely support this as it removes more hidden magic ;-)

👍 Agree.

@gnn If there is some room for wishes I would appreciate a possibility to add lists and dicts:

```python
bus['elec'] = solph.Bus('elec')
bus['coal'] = solph.Bus('coal')

energy_system.add(bus)
```

Just nice to have 😄

**Comment by uvchik** [comment]
@gnn Will you do this for the `v0.2.0` release? Do you have a roadmap or a deadline.

I could help adapting examples, tests and the documentation but I would like to have a little roadmap. I could make some of these adaptions at the workshop for new developers at the developer meeting.

**Comment by gnn** [comment]
Done. See PR #374.
And your requested features are also (somehow) implemented, because you can now do:

```python
energy_system.add(n1, n2, n3)
energy_system.add(*list_of_nodes)
energy_system.add(*dictionary_of_nodes.values())
```

**Comment by uvchik** [comment]
For me this is solved. Closed in favour of #374
